diff --git a/node_modules/laravel-mix/src/FileCollection.js b/node_modules/laravel-mix/src/FileCollection.js
index 34d399a..d4caaaf 100644
--- a/node_modules/laravel-mix/src/FileCollection.js
+++ b/node_modules/laravel-mix/src/FileCollection.js
@@ -2,7 +2,7 @@ let concat = require('concat');
 let path = require('path');
 let fs = require('fs-extra');
 let babel = require('@babel/core');
-let glob = require('glob');
+let globModule = require('glob');
 let _ = require('lodash');
 let { promisify } = require('util');
 let Log = require('./Log');
@@ -88,7 +88,9 @@ class FileCollection {
                 : file
         })
 
-        let globAsync = promisify(glob)
+        let globFn =
+            typeof globModule === 'function' ? globModule : globModule.glob;
+        let globAsync = globFn;
 
         // 3. Expand globs
         let groups = await Promise.all(files.map(async file => {
diff --git a/node_modules/laravel-mix/src/components/JavaScript.js b/node_modules/laravel-mix/src/components/JavaScript.js
index 3c968a2..bc36493 100644
--- a/node_modules/laravel-mix/src/components/JavaScript.js
+++ b/node_modules/laravel-mix/src/components/JavaScript.js
@@ -1,4 +1,4 @@
-const glob = require('glob');
+const globModule = require('glob');
 const File = require('../File');
 const Assert = require('../Assert');
 const { Component } = require('./Component');
@@ -24,7 +24,11 @@ module.exports = class JavaScript extends Component {
      */
     register(entry, output) {
         if (typeof entry === 'string' && entry.includes('*')) {
-            entry = glob.sync(entry);
+            const globSync =
+                typeof globModule.sync === 'function'
+                    ? globModule.sync
+                    : globModule.globSync;
+            entry = globSync(entry);
         }
 
         Assert.js(entry, output);
diff --git a/node_modules/laravel-mix/src/tasks/FileGlob.js b/node_modules/laravel-mix/src/tasks/FileGlob.js
index b790f4e..fedfe59 100644
--- a/node_modules/laravel-mix/src/tasks/FileGlob.js
+++ b/node_modules/laravel-mix/src/tasks/FileGlob.js
@@ -1,9 +1,10 @@
 let path = require('path');
-let glob = require('glob');
+let globModule = require('glob');
 let File = require('../File');
 let { promisify } = require('util');
 let { concat } = require('lodash');
-let globAsync = promisify(glob);
+let globFn = typeof globModule === 'function' ? globModule : globModule.glob;
+let globAsync = globFn;
 
 /** @internal */
 module.exports.FileGlob = class FileGlob {
